### DISCUSSION TITLE
How can I implement "warmup" for a rate-limiter?

### DISCUSSION THREAD

**Discussion by karpoftea** [vladimir-bukhtoyarov/bucket4j#271] [Q&A]
Hi, I want a client to gradually increase load to a certain limit (real world example: gradually increase http-request rate to a cloud service in order to lower client timeouts while service is scaling under load) which is similar to [guava ratelimiter](https://github.com/google/guava/blob/master/guava/src/com/google/common/util/concurrent/RateLimiter.java#L164)(i.e. [SmoothWarmingUp](https://github.com/google/guava/blob/4df79d565eef5e523c04bb6865d93f62c52e98d1/android/guava/src/com/google/common/util/concurrent/SmoothRateLimiter.java#L205)). The problem of guavas implementation that initial rate (cold state) is 1/3 of limit, and its not configurable.
Is there a way to configure warmup in bucket4j?

**Comment by vladimir-bukhtoyarov** [comment]
Hello @karpoftea 

There are no functionality like warm-up. There is only ability to [specify initial amount of tokens](https://bucket4j.com/7.5.0/toc.html#specifying-initial-amount-of-tokens) which is not direct equivalent of Guava's warmup.
